### PR TITLE
Train from scratch

### PR DESCRIPTION
**Pull request by brandenchan**
This branch implements language model training from scratch

### CODE REVIEW

**Comment by brandenchan** [comment]
Currently the models are not able to overfit to even small datasets

**Comment by tholor** [comment]
This might be an issue due to parameter initialization or it's just the normal model behaviour in this combination of dataset, drop_out, low learning rate etc.
We should try to run the same training using google's original script and compare if that one actually overfits.

**Comment by tholor** [comment]
Closing as this was already merged as a part of #188
